server/src/main/java/org/apache/iotdb/db/pipe/processor/PipeDoNothingProcessor.java

```diff
@@ -44,17 +49,22 @@ public void validate(PipeParameterValidator validator) {
   @Override
   public void customize(
       PipeParameters parameters, PipeProcessorRuntimeConfiguration configuration) {
-    // do nothing
+    database =
+        StorageEngine.getInstance()
+            .getDataRegion(
+                (DataRegionId)
+                    ConsensusGroupId.Factory.createFromTConsensusGroupId(
+                        ((PipeTaskRuntimeEnvironment) configuration.getRuntimeEnvironment())
+                            .getRegionId()))
+            .getDatabaseName();
   }
 
   @Override
   public void process(TabletInsertionEvent tabletInsertionEvent, EventCollector eventCollector)
       throws IOException {
     if (tabletInsertionEvent instanceof EnrichedEvent) {
       final EnrichedEvent enrichedEvent = (EnrichedEvent) tabletInsertionEvent;
-      if (enrichedEvent
-          .getPattern()
-          .equals(PipeCollectorConstant.COLLECTOR_PATTERN_DEFAULT_VALUE)) {
+      if (enrichedEvent.getPattern().equals(database)) {
```

Comment:
if (database.startsWith(enrichedEvent.getPattern())) ?

Reply:
I will fix it.